source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -683,7 +727,15 @@ namespace
                 }
                 else if (elem->getObjType() == RtObjType::NODE)
                 {
-                    writeNode(elem->asA<PvtNode>(), doc);
+                    PvtNode* node = elem->asA<PvtNode>();
+                    NodePtr mxNode = writeNode(node, doc);
+                    const PvtNodeDef* nodedef = node->getNodeDef()->asA<PvtNodeDef>();
+                    if (writeOptions && writeOptions->materialWriteOp != RtWriteOptions::MaterialWriteOp::NONE &&
+                        writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::WRITE &&
+                        nodedef->numOutputs() == 1 && nodedef->getPort(0)->getType() == RtType::SURFACESHADER)
```

Comment:
The check for not NONE is already covered by the check for having the WRITE bit set, so that could be removed.